ethcore/src/blockchain/blockchain.rs

```diff
@@ -351,11 +333,13 @@ impl BlockProvider for BlockChain {
 
 	/// Returns numbers of blocks containing given bloom.
 	fn blocks_with_bloom(&self, bloom: &Bloom, from_block: BlockNumber, to_block: BlockNumber) -> Vec<BlockNumber> {
-		let range = from_block as bc::Number..to_block as bc::Number;
-		let chain = bc::group::BloomGroupChain::new(self.blooms_config, self);
-		chain.with_bloom(&range, bloom)
+		// +1, cause it's inclusive range
+		(from_block..to_block + 1)
 			.into_iter()
-			.map(|b| b as BlockNumber)
+			.filter_map(|number| self.block_hash(number).map(|hash| (number, hash)))
```

Comment:
Could possibly be done concurrently, no?

Reply:
yes!